src/main/webapp/search-script.js

```diff
@@ -0,0 +1,84 @@
+function countSearchUnits() {
+  const inputElems = document.getElementsByName("search-units");
+  const arr = []
+  for (let i = 0; i < inputElems.length; i++) {       
+    if (inputElems[i].checked){
+      arr.push(inputElems[i].value);
+    }
+  }
+  return arr;
+}
+
+function countAddUnits() {
+  const inputElems = document.getElementsByName("num-units");
+  const arr = []
+  for (let i = 0; i < inputElems.length; i++) {       
+    if (inputElems[i].checked){
+      arr.push(inputElems[i].value);
+    }
+  }
+  return arr;
+}
+
+function showCourses() {
+  const courseResults = document.getElementById("search-results");
+  const courseName = document.getElementById("search-course").value;
+  const profName = document.getElementById("search-prof").value;
+  const termName = document.getElementById("search-term").value;
+  const units = countSearchUnits();
+  const school = getUserSchool();
+  courseResults.innerHTML = "";
+  const url = new URL("/search", window.location.origin);
+  url.searchParams.set("courseName", courseName);
+  url.searchParams.set("profName", profName);
+  url.searchParams.set("units", units);
+  url.searchParams.set("term", termName);
+  url.searchParams.set("schoolName", getUserSchool())
+  fetch(url)
+  .then(response => response.json())
+  .then((courses) => {
+    courses.forEach(course => courseResults.appendChild(createListElement(course)));
+  });
```

Comment:
Eventually we'll want to add error handling, but that can wait for the polishing stage.